airbyte-integrations/connectors/source-salesforce/source_salesforce/manifest.yaml

```diff
@@ -0,0 +1,112 @@
+version: 5.17.0
+
+type: DeclarativeSource
+
+check:
+  type: CheckStream
+  stream_names:
+    - CalendarView
+
+definitions:
+  streams:
+    custom_object:
+      type: DeclarativeStream
+      name: custom_object
+      $parameters:
+        stream_name: "stream_name"
+      retriever:
+        requester:
+          $ref: "#/definitions/base_requester"
+          path: sobjects/{{parameters["stream_name"]}}/describe
+          http_method: GET
+        record_selector:
+          extractor:
+            type: DpathExtractor
+            field_path: []
+      schema_loader:
+        type: DynamicSchemaLoader
+        retriever:
+          type: SimpleRetriever
+          requester:
+            $ref: "#/definitions/base_requester"
+            path: sobjects/{{parameters["stream_name"]}}/describe
+            http_method: GET
+          record_selector:
+            extractor:
+              type: DpathExtractor
+              field_path: [ ]
+            record_filter:
+              condition: "{{ record['name'] == parameters['stream_name'] }}"
+        schema_type_identifier:
+          schema_pointer: ["fields"]
+          key_pointer: ["name"]
+          type_pointer: ["type"]
+          types_map:
+            - target_type: string
+              current_type: combobox
+  base_requester:
+    type: HttpRequester
+    url_base: https://airbyte--connector.sandbox.my.salesforce.com/services/data/v57.0/
+    authenticator: "#/definitions/authenticator"
+  # TODO: Update authenticator to fetch some additional fields and paste to config
+  authenticator:
+    type: OAuthAuthenticator
+    refresh_request_body: {}
+    token_refresh_endpoint: https://test.salesforce.com/services/oauth2/token
+    grant_type: refresh_token
+    client_id: '{{ config["client_id"] }}'
+    client_secret: '{{ config["client_secret"] }}'
+    refresh_token: '{{ config["refresh_token"] }}'
+
+streams:
+  - $ref: "#/definitions/streams/custom_object"
+
+dynamic_streams:
+  - type: DynamicDeclarativeStream
```

Comment:
One thing I don't see right now is that Salesforce has entities that requires different types of streams. For example, `ContentDocumentLink` will need a parent stream. How do we identify which objects will lead to having a parent stream and others not? 

Or maybe there would be a way to have one stream that always has a parent stream but it can be empty if there is the parameters are not provided. It feels hacky though...

Reply:
To me, this sounds like a layered dependency for the stream template. The structure could be organized by first defining a stream template without parents, then creating one that inherits from it and includes a parent. We can then define two dynamic streams based on a condition (e.g., by name). This way, we can distinguish between streams that require a parent stream and those that do not.